### DISCUSSION TITLE
Getting  -path to work?

### DISCUSSION THREAD

**Discussion by Athelstan-1** [ImageMagick/ImageMagick#6873] [Help]
### ImageMagick version

7.1.1-21

### Operating system, version and so on

Win11

### Description

I'm trying to create thumbnails of my images with the same name as the original file but with the suffix _tn added. The original files are in several levels of subfolders but I want the thumbnails to all be are stored in a specified centralized location.

I can create  the required thumbnails, but they are outputted to the current folder set in the CMD prompt window.

However, I want them to be outputted to the folder "C\1 D\Thumbnails"

I believe -path can be used to set the output folder but haven't been able to successfully add it to my current command.

for /r %a in (*.jpg) do (magick convert "%a" -thumbnail 160x160 -set filename:MYFILE %t_tn.jpg %[filename:MYFILE])

How should the command be modified to get the thumbnails written to the "C\1 D\Thumbnails" folder?

**Comment by snibgo** [comment]
`-path` is only for `magick mogrify`. It is needed there because `magick mogrify` doesn't take an output filename.

For commands that do have an output filename, you can include a full path. For example:
```
-set filename:MYFILE "C\1 D\Thumbnails%t_tn.jpg"
```
I suggest not having spaces in filenames or directory names. Spaces make commands more complicated, because filenames need to be quoted.

I also suggest using just `magick`, not `magick convert`.

EDIT: You may need to escape the backslashes:
```
-set filename:MYFILE "C\\1 D\\Thumbnails%t_tn.jpg"
```

**Reply by Athelstan-1** [reply]
Thanks for your help. Escaping the backslashes was necessary..
The command 

for /r %a in (*.jpg) do (magick "%a" -thumbnail 160x160 -set filename:MYFILE C:\\1_D\\Thumbnails\\%t_th.jpg %[filename:MYFILE])

works for folders with no spaces in their names but the command

for /r %a in (*.jpg) do (magick "%a" -thumbnail 160x160 -set filename:MYFILE "C:\\1 D\\Thumbnails\\%t_th.jpg” %[filename:MYFILE])  

results in the prompt "More?"

Any further suggestions?

**Comment by snibgo** [comment]
> -set filename:MYFILE "C:\1 D\Thumbnails\%t_th.jpg”

You have "smart quotes". The close-quote should be the same character as the open-quote.

And you should escape the backslashes.

**Comment by Athelstan-1** [comment]
Thanks again.

I had recognised the need for escaped backslashes thanks to your first reply, but cut and pasted an earlier version of the command line without them in my post by mistake.

Correcting the smart quote did the trick and the command now runs perfectly at the command prompt.

for /r %a in (*.jpg) do (magick "%a" -thumbnail 160x160 -set filename:MYFILE "C:\\01 C\\thumbnail images\\%t_th.jpg" %[filename:MYFILE])

If you have time, I have one final question.

The above command line works perfectly  in a simple CMD window but doesn't work in a PowerShell window.
I generally prefer to run command line commands in PowerShell as I find the way it breaks the command line into coloured parts useful in identifying syntax errors.

Can the command be easily re-written to work in PowerShell, or is this more trouble than it is worth?

**Comment by snibgo** [comment]
I don't use Powershell and can't advise, sorry.

In a BAT script, each `%` needs to be doubled `%%`:
```
for /r %%a in (*.jpg) do (magick "%%a" -thumbnail 160x160 -set filename:MYFILE "C:\\01 C\\thumbnail images\\%%t_th.jpg" %%[filename:MYFILE])
```

**Comment by Athelstan-1** [comment]
I'm not a true PowerShell user either. I've never learnt its language but use its window to run CMD commands.

Most CMD commands run perfectly in a PowerShell window but loop commands seem to be an exception.

Thanks again for your help. Signing off now as I have a command that works. and the effort required to get it to run in a PowerShell window is probably greater than switching to  a CMD window whenever I need to run the command.
